capsules/extra/src/tickv.rs

```diff
@@ -270,37 +266,85 @@ impl<'a, F: Flash, H: Hasher<'a, 8>, const PAGE_SIZE: usize> flash::Client<F>
                 }
                 _ => {}
             },
-            Operation::GetKey => match ret {
-                Ok(tickv::success_codes::SuccessCode::Complete)
-                | Ok(tickv::success_codes::SuccessCode::Written) => {
-                    self.operation.set(Operation::None);
-                    self.client.map(|cb| {
-                        cb.get_value_complete(
-                            Ok(()),
-                            self.key_buffer.take().unwrap(),
-                            self.ret_buffer.take().unwrap(),
-                        );
-                    });
-                }
-                Err(tickv::error_codes::ErrorCode::EraseNotReady(_)) | Ok(_) => {}
-                _ => {
-                    self.operation.set(Operation::None);
-                    self.client.map(|cb| {
-                        cb.get_value_complete(
-                            Err(ErrorCode::FAIL),
-                            self.key_buffer.take().unwrap(),
-                            self.ret_buffer.take().unwrap(),
-                        );
-                    });
+            Operation::GetKey => {
+                buf_buffer.map(|buf| {
+                    self.ret_buffer.replace(buf);
+                });
```

Comment:
Generally, the system of `continue_operation()` taking a buffer and then returning it, then this code needing to put that buffer right back, but you've got to make sure you put it in the right place, seems like a bug waiting to happen.

I'm satisfied that you did it correctly for GetKey and AppendKey (I think). And I'm not saying you should rewrite it now, but to me it feels _bad_.